magicctapipe/scripts/lst1_magic/semi_automatic_scripts/check_MAGIC_runs.py

```diff
@@ -0,0 +1,134 @@
+#This script allows us to get information about every MAGIC run ID (and subruns) in files (in a time interval) used for common data analysis (MAGIC1, MAGIC2, LST1). 
+
+#The MAGIC files that can be used for analysis are located in the IT cluster in the following directory:
+#/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}
+
+#In this path, 'tel_id' refers to the telescope ID, which must be either 1 or 2. 'YYYY', 'MM', and 'DD' specify the date.
+
+#In the first step, we have to load a dataframe that contains information about the date, the name of the source, and the range of MAGIC #runs. The file in file_path was generated using the spreadsheet "Common MAGIC LST1 data".
+
+import pandas as pd
+from datetime import datetime, timedelta
+import os
+import re
+
+def fix_lists_and_convert(cell):
+    # Remove brackets to avoid double lists and split on ']['
+    parts = cell.replace('][', ',').strip('[]').split(',')
+    return list(dict.fromkeys(int(item) for item in parts))
+
+def table_magic_runs(df, date_min, date_max):
+    df_selected_data = df.iloc[:, [2, 1, 25]]
+    df_selected_data.columns = ['DATE','source', 'MAGIC_runs']
+    grouped_data = df_selected_data.groupby(['DATE', 'source'])
+    result_table = []
+
+    for (date, source), group in grouped_data:
+        if (date>=date_min and date<=date_max):  
+            runs_combined = group['MAGIC_runs'].sum()
+    
+            result_table.append({
+                'DATE': date,
+                'source': source,
+                'MAGIC runs': runs_combined
+            })
+        
+    result = pd.DataFrame(result_table)
+    result['MAGIC runs'] = result['MAGIC runs'].apply(fix_lists_and_convert)
+    return(result)
+
+def existing_files( tel_id, date, source, magic_runs ):
+
+    magic_runs = str(magic_runs)  
```

Comment:
magic_runs is a single run, not a list of runs, right? then a single name "magic_run" would be better.
In general functions should have a doc string explaining the input and output